### PR TITLE
makefile: match operator, bundle and catalog versioning

### PR DESCRIPTION
**Pull request by snir911**
It seems to me the right approach, cc @gkurz @pmores @bpradipt for confirmation this is legit

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,7 +49,7 @@ ifeq ($(USE_IMAGE_DIGESTS), true)
 endif
 
 # Image URL to use all building/pushing image targets
-IMG ?= $(IMAGE_TAG_BASE):$(VERSION)
```
Comment (bpradipt) [inline Makefile:52]: Do you know if the skeleton generated by the latest operator-sdk makes this change ?
If not, will it break bundle and catalog generation if there is an implicit assumption to use version without `v` prefix ?

File: controllers/openshift_controller.go [hunk 2]
```diff
@@ -1962,17 +1962,17 @@ func (r *KataConfigOpenShiftReconciler) resetInProgressCondition() {
...
 }
 
 func (r *KataConfigOpenShiftReconciler) isUpdating() bool {
 	cond := r.findInProgressCondition()
 	if cond == nil {
-		return false;
+		return false
 	}
-	return cond.Status == corev1.ConditionTrue && cond.Reason == "Updating";
+	return cond.Status == corev1.ConditionTrue && cond.Reason == "Updating"
```
Comment (bpradipt) [inline controllers/openshift_controller.go:1975]: /lgtm for this change.

File: Makefile (hunk 1 above)
Comment (snir911) [reply Makefile:52]: it seems the generated skeleton suggests to push this image to "latest", hence, i assumed this was changed locally and it's fine to change it again

Comment (gkurz) [reply Makefile:52]: Operator-sdk introduced the `v` in the bundle and catalog versioning in v1.6.0 with https://github.com/operator-framework/operator-sdk/pull/4406. Specifically this intermediate change in the PR https://github.com/operator-framework/operator-sdk/compare/9ffbb583609404fa3ab27d59c1e1c624255ad8aa..f8a08b782a90e7a1af8493bdaa9693bc61692dd9#diff-97e56898a92a31cc551cc061b2eb64809ad7fc45c2bacd13c660ae207632538fL139

I've checked the review comments that preceded this change but I don't find any suggestion about the extra `v`... Maybe worth asking operator-sdk folks to sort this out.

**Review by gkurz — COMMENTED** [review]
> This will also allow convenient using of IMG_TAG_BUILD for custom builds

This variable doesn't exist in the tree. Assuming you are actually referring to `IMAGE_TAG_BASE`, I'm not sure to see how this change makes things more convenient. Could you elaborate @snir911 ?

File: Makefile (hunk 1 above)
Comment (gkurz) [reply Makefile:52]: I could chat with an operator-sdk developer on [slack](https://kubernetes.slack.com/archives/C017UU45SHL/p1700562561803609). The outcome is that the `v` was introduced for _convenience_, for whatever that means, and we can do pretty much what we want.

**Review by gkurz — APPROVED** [review]
/lgtm
Thanks @snir911 !

File: Makefile (hunk 1 above)
Comment (gkurz) [reply Makefile:52]: @bpradipt can we merge this change ?

Comment (bpradipt) [reply Makefile:52]: sure..

**Review by bpradipt — APPROVED** [review]
/lgtm

**Pull request merged**